src/loaders/Loader.ts

```diff
@@ -12,11 +12,12 @@ import { removeEvent, hasSizeAttribute, hasLoadingAttribute, addEvent } from "..
 export default abstract class Loader<T extends HTMLElement = any> extends Component<ImReadyLoaderEvents> {
   public static EVENTS: string[] = [];
   public options!: ImReadyLoaderOptions;
+  public isReady = false;
+  public isPreReady = false;
+  public hasDataSize = false;
+  public hasLoading = false;
```

Comment:
If you have changed it for testing, how about using [getter](https://stackoverflow.com/a/35991491)?